onsp_co2/models/carbon_factor.py

```diff
@@ -70,6 +71,16 @@ def _compute_child_qty(self):
         for factor in self:
             factor.child_qty = len(factor.child_ids)
 
+    @api.depends('child_ids.descendant_ids')
+    def _compute_descendant_ids(self):
+        for factor in self:
+            if factor.child_ids:
+                for child in factor.child_ids:
+                    factor.descendant_ids |= child
+                    factor.descendant_ids |= child.descendant_ids
+            else:
+                factor.descendant_ids = False
+    
```

Comment:
```suggestion
    @api.depends('child_ids.descendant_ids')
    def _compute_descendant_ids(self):
        for factor in self:
            factor.descendant_ids = factor.child_ids | factor.child_ids.descendant_ids    
```

Wouldn't that work?
Also, it's better to avoid making an assignment several times because it triggers update() and the methods that depend on it just as many times (for example, if a compute depends on descendant_ids, it would be triggered 2 times in your method, which isn't great in terms of perf).